syncer/client.go

```diff
@@ -161,6 +161,8 @@ func (m *syncPeerClient) GetConnectedPeerStatuses() []*NoForkPeer {
 			status, err := m.GetPeerStatus(peerID)
 			if err != nil {
 				m.logger.Warn("failed to get status from a peer, skip", "id", peerID, "err", err)
+
```

Comment:
What do you think about this solution? (one lock less) :D
```
func (m *syncPeerClient) GetConnectedPeerStatuses() []*NoForkPeer {
	var (
		ps        = m.network.Peers()
		syncPeers = make([]*NoForkPeer, len(ps))
		wg        sync.WaitGroup
	)

	for i, p := range ps {
		wg.Add(1)

		go func(peerID peer.ID, pointer **NoForkPeer) {
			defer wg.Done()

			status, err := m.GetPeerStatus(peerID)
			if err != nil {
				m.logger.Warn("failed to get status from a peer, skip", "id", peerID, "err", err)
			} else {
				*pointer = status
			}
		}(p.Info.ID, &syncPeers[i])
	}

	wg.Wait()

	return syncPeers
}```

Also it would be good if you write a test case for this

Reply:
you are right, my mistake :)

Reply:
I'll approve, just make some test